drivers/softlayer/driver.go

```diff
@@ -305,11 +313,41 @@ func (d *Driver) Create() error {
 		for {
 			s, err := d.GetState()
 			if err != nil {
+				log.Debugf("Failed to GetState - %+v", err)
 				continue
 			}
 
 			if s == state.Running {
 				break
+			} else {
+				log.Debugf("Still waiting - state is %s...", s)
+			}
+			time.Sleep(2 * time.Second)
+		}
+	}
+
+	waitForSetupTransactions := func() {
```

Comment:
Any particular reason this is done for using a closure here?  I'd really prefer to move this and the other closures into real methods / helper functions.